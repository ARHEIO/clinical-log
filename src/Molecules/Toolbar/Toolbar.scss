.navigation-toolbar {
  align-items: center;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin: 0 auto;
  max-width: 500px;
}
