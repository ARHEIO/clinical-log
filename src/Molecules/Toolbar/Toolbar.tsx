import React, { ReactElement } from 'react';
import Button from '@material-ui/core/Button';
import './Toolbar.scss';

type Location = 'newsfeed' | 'clinic';

const Toolbar = (props: {location: Location}): ReactElement => {
  const { location } = props;
  return (
    <nav className="navigation-toolbar">
      <Button variant="contained" color="primary" href="#/newsfeed" disabled={location === 'newsfeed'}>Newsfeed</Button>
      <Button variant="contained" color="primary" href="#/clinic" disabled={location === 'clinic'}>Clinical Log</Button>
    </nav>
  );
};
export default Toolbar;
