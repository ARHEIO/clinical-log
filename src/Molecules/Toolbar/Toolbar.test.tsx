import React from 'react';
import { render, RenderResult } from '@testing-library/react';
import pretty from 'pretty';
import Toolbar from './Toolbar';

const renderComponent = (location: any): RenderResult | any => {
  const result = render(
    <Toolbar location={location} />,
  );
  return { ...result };
};

describe('Toolbar', () => {
  it('should render when clinic prop passed', () => {
    const { container } = renderComponent('clinic');
    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });

  it('should render when newsfeed prop passed', () => {
    const { container } = renderComponent('newsfeed');
    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });

  it('should render when bad prop passed', () => {
    const { container } = renderComponent('asdfasdf');
    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });
});
