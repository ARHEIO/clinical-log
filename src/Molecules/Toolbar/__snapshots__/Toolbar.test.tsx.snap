// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar should render when bad prop passed 1`] = `
"<div class=\\"card-container\\">
  <nav class=\\"navigation-toolbar\\"><a class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary\\" tabindex=\\"0\\" aria-disabled=\\"false\\" href=\\"#/newsfeed\\"><span class=\\"MuiButton-label\\">Newsfeed</span><span class=\\"MuiTouchRipple-root\\"></span></a><a class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary\\" tabindex=\\"0\\" aria-disabled=\\"false\\" href=\\"#/clinic\\"><span class=\\"MuiButton-label\\">Clinical Log</span><span class=\\"MuiTouchRipple-root\\"></span></a></nav>
</div>"
`;

exports[`Toolbar should render when clinic prop passed 1`] = `
"<div class=\\"card-container\\">
  <nav class=\\"navigation-toolbar\\"><a class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary\\" tabindex=\\"0\\" aria-disabled=\\"false\\" href=\\"#/newsfeed\\"><span class=\\"MuiButton-label\\">Newsfeed</span><span class=\\"MuiTouchRipple-root\\"></span></a><a class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled\\" tabindex=\\"-1\\" aria-disabled=\\"true\\" href=\\"#/clinic\\"><span class=\\"MuiButton-label\\">Clinical Log</span></a></nav>
</div>"
`;

exports[`Toolbar should render when newsfeed prop passed 1`] = `
"<div class=\\"card-container\\">
  <nav class=\\"navigation-toolbar\\"><a class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled\\" tabindex=\\"-1\\" aria-disabled=\\"true\\" href=\\"#/newsfeed\\"><span class=\\"MuiButton-label\\">Newsfeed</span></a><a class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary\\" tabindex=\\"0\\" aria-disabled=\\"false\\" href=\\"#/clinic\\"><span class=\\"MuiButton-label\\">Clinical Log</span><span class=\\"MuiTouchRipple-root\\"></span></a></nav>
</div>"
`;
